### PR TITLE
Novice Python: Publishing

### LINKED ISSUE #221: Novice Python: Publishing
I will have a stab at this. I think the intro to Jupyter will go in here as well and then we can split later if needed.

### PR DESCRIPTION
**Pull request by joelostblom**
I have adopted Charlotte's R material to Python. Still need to change the exercises and potentially add a jupytext/jupyter section (might do that in a separate PR).

close #221

### CODE REVIEW

**Comment by joelostblom** [comment]
This is ready for review, I will just update with a few links to other sections (like jupytext) when they are written. Maybe @cwickham do you want to review since I have adapted your material? Or do you prefer if I ping one of the Python people?

**Comment by cwickham** [comment]
I'll take a stab at it, but it may benefit from a Python reviewer too.

File: py-publishing.Rmd [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+
+## Questions {#py-publishing-questions}
+
+```{r, child="questions/py-publishing.md"}
+```
+
+## Why should I share my work on the internet?
+
+A key part of any project is communicating what you have learned.
+You've already seen how to create documents to communicate your work with R Markdown,
```
Comment (cwickham) [inline py-publishing.Rmd]: R Markdown? Replace with python equivalent, is it Jupyter Notebooks in this instance?

File: py-publishing.Rmd [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+
+In reverse order this process also describes what you need to do to get your own website online:
+
+1. You need an HTML file that describes what people should see on your page.
+2. You need to host the HTML file on a computer on the internet.
+3. You need a way to associate a URL with the address of your host.
+
+In this chapter,
+you'll learn a process for getting your work online that leverages what you already know ---
+creating HTML files using R Markdown (#1 above) and how to host your work in a GitHub repository (#2).
```
Comment (cwickham) [inline py-publishing.Rmd]: What do they already know?  Sounds on the Python side they haven't seen how to create HTML yet?

File: py-publishing.Rmd [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+`http://{{your_username}}.github.io/{{repo_name}}`.
+
+## How do I get my work on the web?
+
+### A starting point
+
+TODO: revisit this later when more content is fleshed out.  Maybe there will be a repo we can rely on all learners having that we can start from.
+
+In practice you'll probably start thinking about a website once you've already done a lot of work on a project ---
+your project will already have some analysis documented in R Markdown,
```
Comment (cwickham) [inline py-publishing.Rmd]:
```suggestion
your project will already have some analysis documented in a JupyterLab Notebook,
```

File: py-publishing.Rmd [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+by following illustrated steps in [the GitHub guide][github-new-repo].
+
+[github-new-repo]: https://help.github.com/en/articles/create-a-repo
+
+Open JupyterLab and create a new Jupyter Notebook
+("File -> New -> Notebook", or the Python symbol on the launcher page)
+Rename this file to `report.ipynb` and pair it with an `.Rmd` file using Jupytext.
+The pairing to an `.Rmd` file is not necessary,
+but enables a more human-friendly version control workflow.
+
```
Comment (cwickham) [inline py-publishing.Rmd]: Oh now I'm confused.  Do learners need to know about R Markdown, or is this just a convenient way to have the notebook in version control?  What does this mean for my earlier comments about "R markdown"?  Is this something learners will have / need to see earlier (e.g. the R book covers it in "Reproducibility")?

File: py-publishing.Rmd [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+In the first cell create a markdown title that reads "Sharing work".
```
Comment (cwickham) [inline py-publishing.Rmd]: You've written this like learners know what a "markdown title" is...do they?

File: py-publishing.Rmd (hunk 1 above)
Comment (joelostblom) [reply py-publishing.Rmd]: Hmmm, I thought I replaced all these, apparently I missed three!

File: py-publishing.Rmd (hunk 2 above)
Comment (joelostblom) [reply py-publishing.Rmd]: Good catch! Reworded.

File: py-publishing.Rmd (hunk 5 above)
Comment (joelostblom) [reply py-publishing.Rmd]: Yes, this is included in the Jupyter section I just made, but I reworded to clarify here also.

File: py-publishing.Rmd (hunk 4 above)
Comment (joelostblom) [reply py-publishing.Rmd]: This is just a convenient way of having notebooks under version control. It is what we recommend in the Jupyter section that I have in another PR, so although it is not technically necessary here, I kept it as .Rmd for consistency. Your earlier comments about R Markdown are not influenced by this.

**Comment by joelostblom** [comment]
Thanks @cwickham ! I updated according to your feedback.

**Pull request merged**